subsys/bluetooth/mesh/friend.c

```diff
@@ -60,12 +60,12 @@ struct friend_pdu_info {
 	uint32_t  iv_index;
 };
 
-BUILD_ASSERT(CONFIG_BT_MESH_LABEL_COUNT <= 0xFFFU, "Friend doesn't support more than 4096 labels.");
+BUILD_ASSERT(CONFIG_BT_MESH_LABEL_COUNT <= 0xFFFU, "Friend doesn't support more than 4095 labels.");
```

Comment:
This is a build assert. I don't think it should be removed because code will silently fail if someone should modify the kConfig option, or override that option in application level KConfig file. Of course it could be debated how far we can go, but the reasons for build assert is the  struct in `friend_adv` has `uint16_t uuidx:15`, which can store maximum of 4096 (0 to 4095) indices.

Reply:
I agree with @Thalley, we do not need to check values after KConfig. BUILD_ASSERT is kind of useless here. If somebody changes boundaries on his own apart suggested, this is his responsibility to follow up correctness of it.

Reply:
> This is a build assert. I don't think it should be removed because code will silently fail if someone should modify the kConfig option, or override that option in application level KConfig file.

If they modify the Kconfig option, they can also just modify the build assert, so I don't believe that's a good argument. 

If an application choose to override the Kconfig option at the application level, they also kind of just go into a weird state where they can't assume any correctness anymore. 

I'm OK with keeping this build assert, but it seems useless when it's covered by Kconfig.

Reply:
Ok, so then the suggestion for Lingao is to revert the change.